### PR TITLE
Added 1.18 and 1.19 advancements, logic tweaks, new settings

### PR DESCRIPTION
**Pull request by AkumaGath17**
May need some tests.
(Known issue: Two by two currently on AP doesn't need Bucket on logic, but it should)

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
 	"name": "Minecraft Randomizer",
 	"game_name": "Minecraft",
 	"platform": "PC",
-	"package_version": "0.1.0",
+	"package_version": "0.2.0",
 	"package_uid": "minecraft_randomizer_emotracker-neomatamune",
-	"author": "Neomatamune, Cyb3R",
+	"author": "Neomatamune, Cyb3R (AkumaGath17#0725 Edit)",
```
Comment (Cyb3RGER) [inline manifest.json:7]: Please just add your name to the list like this:
```suggestion
	"author": "Neomatamune, Cyb3R, AkumaGath17#0725",
```

File: README.md [hunk 2]
```diff
@@ -16,4 +16,10 @@ The pack does not autotrack the village and pillager outpost structures, as thes
...
 ## Known Issues
 
-The pack doesn't track yet if you have enough advancements and/or dragon egg shards to reach the goal or kill the ender dragon. Therefore postgame locations might appear as "in logic" when in reality you can reach them yet as you are still missing the nessaracy advancements and/or dragon egg shards
+The pack doesn't track yet if you have enough advancements and/or dragon egg shards to reach the goal or kill the ender dragon. Therefore postgame locations might appear as "in logic" when in reality you can reach them yet as you are still missing the nessaracy advancements and/or dragon egg shards.
+
+## About 0.2.0 new features
+
+When using "Show Excluded: Off" use temporarily as a goal "None" to hide every Postgame advancement, while not using "None" as goal toggle "Include postgame advancements:On/Off" to check when these are in logic.
+If you want to exclude an advancement in particular (because you also can on the .yaml from Archipelago) just go to: 
+/locations/locations.json and comment the advancement's visibility_rules except "showexcluded_on", these will keep appearing when using "Show Excluded: On".
```
Comment (Cyb3RGER) [inline README.md:25]: Maybe we can reach out to AP devs and get them to add excluded locations in slot data so we can automate this? Not a required change but an idea.

File: locations/locations.json [hunk 3]
```diff
@@ -51,10 +62,14 @@
                 ],
                 "sections": [
                     {
-                        "name": "Getting an Upgrade",
+                        "name": "Have a stone pickaxe in your inventory.",
                         "access_rules": [
                             "$tools_cobble",
-                            "village, [$tools_cobble]"
```
Comment (Cyb3RGER) [inline locations/locations.json:57]: why change the access_rules here to never be yellow? (also see Isn't it Iron Pick)

File: locations/locations.json [hunk 4]
```diff
@@ -128,10 +158,14 @@
                 ],
                 "sections": [
                     {
-                        "name": "Isn't it Iron Pick",
+                        "name": "Have an iron pickaxe in your inventory.",
                         "access_rules": [
-                            "$tools_iron, [$has_iron]",
-                            "$village, [$tools_iron], [$has_iron]"
+                            "[$tools_iron], [$has_iron]",
```
Comment (Cyb3RGER) [inline locations/locations.json:163]: why change access_rules here to be always yellow? (also see Getting an Upgrade)

File: locations/locations.json [hunk 5]
```diff
@@ -226,9 +280,14 @@
                 ],
                 "sections": [
                     {
-                        "name": "Cover Me With Diamonds",
+                        "name": "Have any type of diamond armor in your inventory.",
                         "access_rules": [
-                            "[$armor_diamond], @Overworld/Diamonds!/Diamonds!"
+                            "[$armor_diamond], @Overworld/Diamonds!/Consigue diamante"
```
Comment (Cyb3RGER) [inline locations/locations.json:285]: wrong location name:
```suggestion
                            "[$armor_diamond], @Overworld/Diamonds!/Have a diamond in your inventory."
```

File: locations/locations.json [hunk 6]
```diff
@@ -658,12 +820,14 @@
                 ],
                 "sections": [
                     {
-                        "name": "Uneasy Alliance",
+                        "name": "Kill a ghast while in the overworld.",
                         "access_rules": [
-                            "$nether, $has_diamond_pickaxe, [$rod], [$include_hard]",
-                            "$nether, $village, [$has_diamond_pickaxe], [$rod], [$include_hard]",
-                            "$nether, $city, [$has_diamond_pickaxe], [$rod], [$include_hard]",
-                            "$nether, $bastion, [$has_diamond_pickaxe], [$rod], [$include_hard]"
```
Comment (Cyb3RGER) [inline locations/locations.json:666]: Why the change to the access_rules here? (Same as Getting an Upgrade)

File: locations/locations.json [hunk 7]
```diff
@@ -779,10 +992,18 @@
...
                     {
-                        "name": "Who is Cutting Onions?",
+                        "name": "Have a block of crying obsidian in your inventory.",
                         "access_rules": [
                             "nether, [$can_piglin_trade]",
-                            "bastion, [$can_piglin_trade]"
+                            "bastion, [$can_piglin_trade]",
+                            "$tools_diamond, [$can_piglin_trade]",
+                            "$city, [$can_piglin_trade]",
+                            "$village, [$can_piglin_trade]",
```
Comment (Cyb3RGER) [inline locations/locations.json:1001]: how do city, village and diamond tools affect access here?

File: locations/locations.json [hunk 8]
```diff
@@ -820,9 +1046,17 @@
...
                     {
-                        "name": "Hidden in the Depths",
+                        "name": "Have ancient debris in your inventory.",
                         "access_rules": [
-                            "$bastion, [$nether], [$can_brew_potions], [$bed], [$has_diamond_pickaxe]"
+                            "$bastion, [$nether], [$can_brew_potions], [$bed], [$has_diamond_pickaxe]",
+                            "$nether, $village, [$can_brew_potions], [$bed], [$has_diamond_pickaxe]",
+                            "$nether, $city, [$can_brew_potions], [$bed], [$has_diamond_pickaxe]",
+                            "$nether, $tools_diamond, [$can_brew_potions], [$bed], [$has_diamond_pickaxe]",
+                        ],
```
Comment (Cyb3RGER) [inline locations/locations.json:1055]: how do city, village and diamond tools affect access here?

File: locations/locations.json [hunk 9]
```diff
@@ -839,10 +1073,16 @@
...
                 "sections": [
                     {
-                        "name": "Country Lode, Take Me Home",
+                        "name": "Use a compass on a lodestone.",
                         "access_rules": [
-                            "[$nether], @Nether/Hidden in the Depths/Hidden in the Depths, [$has_gold]",
-                            "[$nether], $scraps, [@Nether/Hidden in the Depths/Hidden in the Depths], [$has_gold]"
+                            "[$nether], @Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris, $tools_diamond, [$has_gold]",
+                            "[$nether], $scraps, $tools_diamond, [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$has_gold]",
+                            "[$nether], $bastion, [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$has_gold]"
```
Comment (Cyb3RGER) [inline locations/locations.json:1080]: Wrong location names here

File: locations/locations.json [hunk 10]
```diff
@@ -897,9 +1147,16 @@
                 ],
                 "sections": [
                     {
-                        "name": "Cover Me in Debris",
+                        "name": "Have a full set of netherite armor in your inventory.",
                         "access_rules": [
-                            "[$nether], [$armor_diamond], [$has_all_scraps], [$crafting_ingots], [@Overworld/Diamonds!/Diamonds!], @Nether/Hidden in the Depths/Hidden in the Depths, [$include_hard]"
+                            "$scraps, $tools_diamond, [$nether], [$armor_diamond], [$has_all_scraps], [$crafting_ingots], [@Overworld/Diamonds!/Consigue diamante], [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$include_hard]",
+                            "$bastion, [$nether], [$armor_diamond], [$has_all_scraps], [$crafting_ingots], [@Overworld/Diamonds!/Consigue diamante], [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$include_hard]",
+                            "[$nether], @Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris, $tools_diamond, [$armor_diamond], [$has_all_scraps], [$crafting_ingots], [@Overworld/Diamonds!/Consigue diamante], [$include_hard]"
```
Comment (Cyb3RGER) [inline locations/locations.json:1154]: Wrong location names here

File: locations/locations.json [hunk 11]
```diff
@@ -1290,10 +1651,39 @@
...
+                        "map": "all",
+                        "x": 738,
+                        "y": 654
+                    }
+                ],
+                "sections": [
+                    {
+                        "name": "Use a jukebox while in a meadow.",
+                        "access_rules": [
+                            "[$basic_combat], [@Overworld/Diamonds!/Consigue diamante], [$include_hard]"
```
Comment (Cyb3RGER) [inline locations/locations.json:1680]: wrong location name here

File: locations/locations.json [hunk 12]
```diff
@@ -1501,10 +1966,39 @@
                 ],
                 "sections": [
                     {
-                        "name": "Very Very Frightening",
+                        "name": "Hit a villager with a channeling trident on a thunderstorm.",
                         "access_rules": [
                             "$village, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]",
-                            "@Overworld/Zombie Doctor/Zombie Doctor, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]"
+                            "@Overworld/Zombie Doctor/Cura a un zombie aldeano con:\n-Lanza pocion de debilidad\n-Manzana Dorada, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]"
```
Comment (Cyb3RGER) [inline locations/locations.json:1972]: wrong location name here

File: locations/locations.json [hunk 13]
```diff
@@ -1654,10 +2184,62 @@
                 ],
                 "sections": [
                     {
-                        "name": "Surge Protector",
+                        "name": "Protect a villager from a lightning bolt without starting a fire.",
                         "access_rules": [
                             "$village, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]",
-                            "@Overworld/Zombie Doctor/Zombie Doctor, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]"
+                            "@Overworld/Zombie Doctor/Cura a un zombie aldeano con:\n-Lanza pocion de debilidad\n-Manzana Dorada, [$channeling], [$can_use_anvil], [$can_enchant], [$include_hard]"
```
Comment (Cyb3RGER) [inline locations/locations.json:2190]: wrong location name here

File: locations/locations.json [hunk 14]
```diff
@@ -1736,10 +2333,16 @@
...
                 "sections": [
                     {
-                        "name": "Serious Dedication",
+                        "name": "Have a netherite hoe in your inventory.",
                         "access_rules": [
-                            "[$nether], @Nether/Hidden in the Depths/Hidden in the Depths, [$has_gold]",
-                            "[$nether], $scraps, [@Nether/Hidden in the Depths/Hidden in the Depths], [$has_gold]"
+                            "[$nether], @Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris, $tools_diamond, [$has_gold]",
+                            "[$nether], $scraps, $tools_diamond, [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$has_gold]",
+                            "[$nether], $bastion, [@Nether/Hidden in the Depths/Consigue el bloque de Ancient Debris], [$has_gold]"
```
Comment (Cyb3RGER) [inline locations/locations.json:2340]: wrong location names here

File: locations/locations.json [hunk 15]
```diff
@@ -1908,9 +2623,62 @@
...
+                        "map": "all",
+                        "x": 822,
+                        "y": 1158
+                    }
+                ],
+                "sections": [
+                    {
+                        "name": "Have an Allay drop a Cake at a Note Block.",
+                        "access_rules": [
+                            "@Archipelago/The Lie/Come tarta, [$tools_iron], [$has_iron], [$outpost]"
```
Comment (Cyb3RGER) [inline locations/locations.json:2676]: wrong location name here

**Review by Cyb3RGER — COMMENTED** [review]
Left some comments/questions, mainly on access rules.
Maybe double check location name in access rules. Poptracker tells you in the log when it tries to access a location it cannot find.
Would be cool if you could color in the new advancement icons like the old ones.

File: locations/locations.json (hunk 8 above)
Comment (AkumaGath17) [reply locations/locations.json:1055]: -Obtain a diamond pickaxe from village/city loot.
-Obtaining diamonds out of logic (not necessarily with has_iron).
Still you can get diamond pickaxe from villager trades so i should just make it hard locked behind Nether or Bastion.

File: locations/locations.json (hunk 7 above)
Comment (AkumaGath17) [reply locations/locations.json:1001]: Crying obsidian can generate as part of ruined portals.
-Obtain a diamond pickaxe from village/city loot.
-Obtaining diamonds out of logic (not necessarily with has_iron).
Still you can get diamond pickaxe from villager trades so i should make it at least always Yellow along with Zombie Doctor.

File: locations/locations.json (hunk 6 above)
Comment (AkumaGath17) [reply locations/locations.json:666]: You can obtain Obsidian blocks from ruined portals loot chest.

File: locations/locations.json (hunk 4 above)
Comment (AkumaGath17) [reply locations/locations.json:163]: You can technically manage to obtain an iron pickaxe from a mineshaft looting chest, or zombie doctor.

File: locations/locations.json (hunk 3 above)
Comment (AkumaGath17) [reply locations/locations.json:57]: True, not only you can indeed get the stone pickaxe from trades, you can even heal a Zombie Villager at "any" point so guess this one has to be at least Yellow.

**Comment by AkumaGath17** [comment]
Zombie doctor on most of the comments makes me think if i should make those advancements "Red" behind the doctor advancement, but Zombie Doctor also being "Yellow" could be enough even tho it could be confusing to figure it out? Only mentions to certain advancements like Zombie doctor (and basically all those which seems like i ended up forgetting with the translation lol) are in the code to replicate the rules of AP so i think i should just let these in Yellow.

Anyways i gotta change for sure from my fork some advancements like Getting and Upgrade which i got simply wrong and the other stuff you highlighted.

Aaaand as i don't really want to start modifying the PR and im not even sure if that is doable lmao i will just open another one when i can, changing from my current fork commit...

Srry bad english
